### ISSUE TITLE
Should resize_min functions fail?

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#2002]
We have an `resize_min` function for `vector_t` and some other structures.

The documentation says:

> Note that this function involves additional memory allocation and may result an out-of-memory error.

This would happens due to `realloc` returning `0`. But in fact when `realloc` fails, it leave the original piece of memory intact. Thus, even though it has "failed", our vector is still in a usable state. We simply did not manage to release memory, but the program can continue safely. In other words, it is always safe to ignore an error from `resize_min`.

Do we want to:

 - Simply not throw an error, and instead document that this function _attempts_ to release unused memory, without promising success?
 - Alternatively, simply make the documentation sound less scary, and explain that failures can be ignored? Even if future implementations change, they _should_ leave the vector in a consistent state on failure.

**Comment by ntamas** [comment]
I think that `resize_min` can fail only in very exceptional circumstances, and if it _does_ fail, there will potentially be further OOM errors down the line when the user tries to do anything else. I'd handle the failure of `realloc()` gracefully in `resize_min` and make it return `IGRAPH_SUCCESS` unconditionally. The documentation should then be adjusted accordingly. In 0.11 we can make it return `void` if we really want to, although I don't think it's necessary.

**Comment by szhorvat** [comment]
I made the change in fa741e2333e5efcd80b89ea28ba2b5bb413c1e5e, not it always returns `IGRAPH_SUCCESS`.

> I think that `resize_min` can fail only in very exceptional circumstances

In fact, it will not fail under _any_ circumstances. Internally, it just calls `realloc()`. `realloc()` either succeeds and shrinks the storage, or it fails and returns `NULL`. In the latter case, it leaves the original memory chunk intact. Thus, the vector never becomes unusable. At worst, it does not release the unused capacity.

Therefore, I would just change the return type to `void` already for 0.10, as it is now that we are making breaking changes. I did not yet do this in the commit above.
